### ISSUE TITLE
Light-R1-32B复现stage2出现问题

### ISSUE THREAD

**Issue by QingqingSun-Bao** [Qihoo360/Light-R1#42]
基于Qwen2.5-32B-Instruct 模型在stage1阶段的指标和论文上的三个指标相差大概1个点左右，可以认为是完成复现了吧。但是在stage2阶段模型过拟合的很严重，aime指标均在69分以下，
![Image](https://github.com/user-attachments/assets/63d6687d-a4fd-4254-9962-d895a063f3f3)
以下是我在stage2阶段的训练配置，BS保持在32，学习率1e-5
torchrun $DISTRIBUTED_ARGS src/train.py \
    --stage sft \
    --do_train \
    --max_steps -1 \
    --model_name_or_path output/Qwen-stage1-sft-32B-20k-759k/checkpoint-1028 \
    --dataset sft_stage2 \
    --template qwen \
    --finetuning_type full \
    --output_dir output/Qwen-stage2_sft-32B-ckp1028-759k_20k-bs32-lr1e5 \
    --preprocessing_num_workers 16 \
    --sequence_parallel_size 1 \
    --gradient_checkpointing True \
    --flash_attn fa2  \
    --cache_dir .cache \
    --overwrite_cache \
    --cutoff_len 20000 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 2 \
    --lr_scheduler_type cosine \
    --save_strategy epoch \
    --logging_steps 1 \
    --adam_beta1 0.9 \
    --adam_beta2 0.95 \
    --adam_epsilon 1e-8 \
    --max_grad_norm 1.0 \
    --weight_decay 0.1 \
    --warmup_ratio 0.01 \
    --save_total_limit 20 \
    --learning_rate 1e-5 \
    --save_only_model false \
    --num_train_epochs 10 \
    --bf16 true \
    --plot_loss \
    --seed 42 \
    --do_eval false \
    --deepspeed ./examples/deepspeed/ds_z3_config.json \
    --report_to tensorboard \
    --overwrite_output_dir \
    --ddp_timeout 288000000 \
    --enable_liger_kernel

大佬可以帮忙分析下原因吗？

**Comment by HaoshengZou** [comment]
cosine lr, `--num_train_epochs 10` 和100还是不一样的，我们约等于前10个epoch都在constant lr
可是试试写100，测前10个epoch
warmup可以写0
不行再 @juemifuji

**Comment by QingqingSun-Bao** [comment]
> cosine lr, `--num_train_epochs 10` 和100还是不一样的，我们约等于前10个epoch都在constant lr 可是试试写100，测前10个epoch warmup可以写0 不行再 [@juemifuji](https://github.com/juemifuji)

谢谢大佬提醒，我现在有个疑问，就是学习率设置的这么大会不会过拟合？因为训练的loss已经很低了，甚至没有下降的空间了，上面的图片也显示收敛的速度是很快的

**Comment by QingqingSun-Bao** [comment]
@juemifuji 您好，第二阶段的指标按照cosine lr, --num_train_epochs100，还是没有提升，可以帮忙指导以下吗？

**Comment by wjw136** [comment]
麻烦打扰一下，请问使用deepspeed zero3 offload微调32B的模型，cut_off_len设置成2w，不会显存OOM嘛，是还有用什么降低显存的方法吗？
